src/Compilers/CSharp/Portable/Symbols/UnboundGenericType.cs

```diff
@@ -119,4 +119,58 @@ public override int GetHashCode()
                 : Hash.Combine(_name, _errorInfo.Code);
         }
     }
+
+    /// <summary>
+    /// Used for lightweight binding of type constraints. Instead of binding type arguments,
+    /// we'll just use these placeholders instead. That's good enough binding to compute
+    /// <see cref="TypeSymbol.IsValueType"/> with minimal binding.
+    /// </summary>
+    internal sealed class PlaceholderTypeArgumentSymbol : ErrorTypeSymbol
+    {
+        private static readonly TypeWithAnnotations s_instance = TypeWithAnnotations.Create(new PlaceholderTypeArgumentSymbol());
+
+        public static ImmutableArray<TypeWithAnnotations> CreateTypeArguments(ImmutableArray<TypeParameterSymbol> typeParameters)
+        {
+            return typeParameters.SelectAsArray(_ => s_instance);
+        }
+
+        private PlaceholderTypeArgumentSymbol()
+        {
+        }
+
+        protected override NamedTypeSymbol WithTupleDataCore(TupleExtraData newData)
+        {
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return string.Empty;
+            }
+        }
+
+        internal override bool MangleName
+        {
+            get
+            {
+                Debug.Assert(Arity == 0);
+                return false;
+            }
+        }
+
+        internal override DiagnosticInfo? ErrorInfo
+        {
+            get
+            {
+                return null;
+            }
+        }
+
+        internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverrideOpt = null)
```

Comment:
>Equals [](start = 31, length = 6)

Should we override GetHashCode?

Reply:
I removed it based on earlier feedback.

---
In reply to: [503348130](https://github.com/dotnet/roslyn/pull/48270#discussion_r503348130) [](ancestors = 503348130)

Reply:
> I removed it based on earlier feedback.

I think we should have it here.

---
In reply to: [503602842](https://github.com/dotnet/roslyn/pull/48270#discussion_r503602842) [](ancestors = 503602842,503348130)